### PR TITLE
Pass subscription ID to Az CLI commands

### PR DESCRIPTION
**Pull request by KeithJRome**
## All Submissions:
-------------------------------------
* [YES] Have you followed the guidelines in our Contributing [document](./CONTRIBUTING.md)?
* [YES] Have you added an explanation of what your changes do and why you'd like us to include them?
* [NA] I have updated the documentation accordingly.
* [NA] I have added tests to cover my changes.
* [YES] All new and existing tests passed.
* [YES] My code follows the code style of this project.
* [YES] I ran lint checks locally prior to submission.
* [YES] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

## What is the current behavior?
-------------------------------------
Some Azure CLI commands allow for default parameter values for Subscription ID, Resource Group Name, etc. Cobalt is currently not explicitly setting these parameters when the Terraform modules shell out to run direct Az commands. This does not seem to be causing problems on the current build servers (yet) but does have potential for build unpredictability as those parameters will fall back to configurable defaults from the context of the logged-in user.

Issue Number: #290 

## What is the new behavior?
-------------------------------------
Calls to the Azure CLI now explicitly pass the Subscription ID and Resource Group.

## Does this introduce a breaking change?
-------------------------------------
- [NO]

## Any relevant logs, error output, etc?
-------------------------------------
(If it’s long, please paste to https://ghostbin.com/ and insert the link here.)

## Other information
-------------------------------------

### CODE REVIEW

File: infra/modules/providers/azure/app-service/main.tf [hunk 1]
```diff
@@ -39,6 +39,7 @@ resource "azurerm_app_service" "appsvc" {
     linux_fx_version     = format("DOCKER|%s/%s", var.docker_registry_server_url, local.app_configs[count.index].image)
     always_on            = var.site_config_always_on
     virtual_network_name = var.vnet_name
+    ip_restriction       = []
```
Comment (KeithJRome) [inline infra/modules/providers/azure/app-service/main.tf]: FYI, this is done to resolve a spurious error during what should be a clean subsequent APPLY (when no changes were made to schema).

File: infra/templates/az-isolated-service-single-region/ase.tf [hunk 2]
```diff
@@ -147,13 +149,23 @@ resource "null_resource" "auth" {
   }
 
   provisioner "local-exec" {
-    command = "az webapp auth update -g \"$RES_GRP\" -n \"$APPNAME\" --enabled true --action LoginWithAzureActiveDirectory --aad-token-issuer-url \"$ISSUER\" --aad-client-id \"$APPID\""
+    command = <<EOF
```
Comment (TechnicallyWilliams) [inline infra/templates/az-isolated-service-single-region/ase.tf]: Just wanted to say I love this new formatting.  Definitely next level!

**Comment by iphilpot** [comment]
Should we have broken this into smaller chunks of work? This is a large, PR...

**Comment by KeithJRome** [comment]
@iphilpot the rebase from master was messy - I've cleaned up the commits now. This PR wasn't as big as the messy history made it look.

**Comment by KeithJRome** [comment]
Yes, these commits only targeted the inline `local-exec` blocks. I need to add the extra params to those `program` blocks also.

**Pull request merged**